telosnetwork/teloscan
# #107 | Page tabs and routing fix

# Fixes #107

## Description
Route hashes, which are used by the quasar tabbing component, currently break if an invalid route is navigated to. This PR fixes how hashes are handled.

## Test scenarios


- go to https://deploy-preview-263--dev-mainnet-teloscan.netlify.app/address/0xdb45411265782debf29a7b58ab475bf75c9b19f3
    - the page should load fine with the Transaction tab selected
- go to https://deploy-preview-263--dev-mainnet-teloscan.netlify.app/address/0xdb45411265782debf29a7b58ab475bf75c9b19f3#transactions
    - the page should load fine with the Transaction tab selected
- on the same page, click around to the different tabs (ERC20, ERC721, etc)
    - tabs should load as normal
- go to an invalid hash, like https://deploy-preview-263--dev-mainnet-teloscan.netlify.app/address/0xdb45411265782debf29a7b58ab475bf75c9b19f3#fake
    - you should be directed to the transactions tab
- go to the `#contract` tab for an address which has no contract tab, such as https://deploy-preview-263--dev-mainnet-teloscan.netlify.app/address/0xdb45411265782debf29a7b58ab475bf75c9b19f3#contract
    - you should be directed to the transactions tab
- go to the `#contract` tab for an address which DOES have a contract tab, such as https://deploy-preview-263--dev-mainnet-teloscan.netlify.app/address/0xd102ce6a4db07d247fcc28f366a623df0938ca9e#contract
    - the contract tab should be loaded
- go to a transaction page, e.g. https://deploy-preview-263--teloscan-stage.netlify.app/tx/0xbd2510d29472bf0ec2ba9def6709e306a7c8f1e1fb1297e3e18279aa5ca3e77d#general
- click around to different tabs 
    - the tabs should load normally    
- go to https://deploy-preview-263--teloscan-stage.netlify.app/staking (you will need to be signed in for this)
- click around different tabs
    - tabs should load normally

## Checklist:
